pandas/io/json/_json.py

```diff
@@ -726,7 +742,19 @@ def read(self):
             obj = concat(self)
         elif self.lines:
             data = ensure_str(self.data)
-            obj = self._get_object_parser(self._combine_lines(data.split("\n")))
+            if self.nrows:
+                compiled_pattern = re.compile(".*\n")
+                data_iterator = compiled_pattern.finditer(data)
+                data = []
+                nrows_seen = 0
+                for vals in data_iterator:
+                    if nrows_seen >= self.nrows:
+                        break
+                    data.append(vals.group(0))
+                    nrows_seen += 1
```

Comment:
I _think_ the suggestion was to do something like
```python
                data_generator = (vals.group(0) for vals in compiled_pattern.finditer(data))
                data = islice(data_generator, self.nrows)
```
(you'd need to import `islice` from `itertools`), would this work?

Reply:
Thanks for the review.
Wouldn't this load the entire string in memory?
The main reason for using an iterator was to not load the entire string into memory at once.
Please correct me if I am wrong.

Reply:
> Wouldn't this load the entire string in memory?

My understanding is that generators are lazy, so this wouldn't load the entire string in memory. I'll time it tomorrow to check

Reply:
Yeah I also believe generators are lazy but in the snippet you shared the variable data_generator is actually holding a tuple with all the lines. Because it is collecting all the groups from the generator. So, in that case, I was saying that the entire string would be in memory at once.

Reply:
> data_generator is actually holding a tuple 

Are you sure it's holding a tuple? If I do
```python
type(data_generator)
```
in the debugger I get
```python
generator
```

Anyway, I'll check this tomorrow, it's getting late where I live :)

Reply:
My bad.
Thanks a lot.

Reply:
I have made the changes @MarcoGorelli You may review